### PR TITLE
[flutter] ensure vmservice binding is registered in profile mode

### PR DESCRIPTION
**Pull request by jonahwilliams**
Ensure that devtools will launch from the cmdline on a profile build

WIP due to testing

### CODE REVIEW

**Comment by jonahwilliams** [comment]
FYI @Hixie .

**Comment by Hixie** [comment]
Oh, haha, good catch.

**Comment by jonahwilliams** [comment]
I'd like to find a way to test this without adding a devicelab test, but I'm scratching my head a bit right now

File: packages/flutter/lib/src/foundation/binding.dart [hunk 1]
```diff
@@ -246,14 +253,6 @@ abstract class BindingBase {
...
-      registerStringServiceExtension(
-        name: 'connectedVmServiceUri',
-        getter: () async => connectedVmServiceUri ?? '',
-        setter: (String uri) async {
-          connectedVmServiceUri = uri;
-        },
-      );
-
       registerStringServiceExtension(
         name: 'activeDevToolsServerAddress',
```
Comment (kenzieschmoll) [inline packages/flutter/lib/src/foundation/binding.dart:257]: this one probably needs to move out of the assert block too right?

Comment (jonahwilliams) [reply packages/flutter/lib/src/foundation/binding.dart]: Ahh good catch

**Review by kenzieschmoll — APPROVED** [review]

**Comment by jonahwilliams** [comment]
Soliciting testing ideas:

I can't turn off asserts in our unit tests, so I can't show that the extension is still registered.
I can't run a profile build in the tool integration tests, the flutter_tester only supports debug
I could add a new devicelab test to verify devtools connects, but sometimes devtools fails to activate on CI and I'm certain this would be flaky.

**Comment by Hixie** [comment]
I can't think of a way to really test this other than an end-to-end test in devicelab either.

**Comment by Hixie** [comment]
Maybe the first step is to fix the flakiness. :-)

(In the meantime you could also make the test pass in the case where it fails to bootstrap devtools, since we already test that in existing tests.)

**Comment by jonahwilliams** [comment]
Yeah, I think that's what we have to do. if its flaky we can revisit.

**Comment by jonahwilliams** [comment]
I added a devicelab test, will add to infra config after this lands

**Pull request merged**